pkg/config/loader/loader.go

```diff
@@ -819,3 +845,37 @@ func loadInternalClient(l *zerolog.Logger, conf *server.InternalClientTLSConfigF
 		clientv1.WithLogger(l),
 	), nil
 }
+
+// checkDatabaseTimezone validates that the database instance timezone is set to UTC.
+// It creates a temporary connection to check the timezone without using the AfterConnect hook.
+func checkDatabaseTimezone(connConfig *pgx.ConnConfig, dbName string, dbLabel string, l *zerolog.Logger) error {
+	tempConn, err := pgx.ConnectConfig(context.Background(), connConfig)
+	if err != nil {
+		return fmt.Errorf("could not create temporary connection to %s to check timezone: %w", dbLabel, err)
+	}
+	defer tempConn.Close(context.Background())
+
+	var dbTimezone string
+	if err := tempConn.QueryRow(context.Background(), "SHOW timezone").Scan(&dbTimezone); err != nil {
+		return fmt.Errorf("could not query %s timezone: %w", dbLabel, err)
+	}
+
+	// Accept both "UTC" and "Etc/UTC" as valid UTC timezones
+	if dbTimezone != "UTC" && dbTimezone != "Etc/UTC" {
```

Comment:
Is this case sensitive/do we have a link to these options in the PG docs or source code?